### PR TITLE
Adding AWS lambda blueprint

### PR DESCRIPTION
**Pull request by xdanw**
In development. Tested to successfully add CI items; does not connect the function to API gateway.

**General**
 - [ ] Your branch is up-to-date with development branch
 - [ ] A Readme.md file is generated as part of the blueprint which, at the minimum, explains how to use the blueprint after it is generated(like adding any missing steps, creating accounts, setting up docker containers, applying the yaml using XL-CLI, running release etc). Place this file preferably in the XebiaLabs folder. Do not add the blueprint generation steps here as it wont be useful since the file is ceated only after the blueprint generation.
 - [ ] The blueprint must use an XL product at some capacity
 - [ ] There are no sensitive info in blueprints(tokens, credentials, licence), for questions that ask for password or sensitive info use `secret: true` in the `bleuprint.yaml` parameter definition.
 - [ ] Do not add the `xebialabs/.gitignore`, `xebialabs/values.xlvals` and `xebialabs/secrets.xlvals` to the `bleuprint.yaml` files section, and do not check in those files, as they are autogenerated based on parameter definitions. Secrets are created when a parameter in the `blueprint.yaml` has the `secret: true` directive. For adding parameters to `values.xlvals` use the `saveInXlVals: true` directive.
 - [ ] Please do not ask questions for trivial things like phase names, task names, folder names etc, ask questions that add value and be opinionated where possible. For example ask for a project name and derive folder names, task names etc based on that.
 - [ ] Folder and file names must use dash-case, for example: `aws/sample-app-demo`, `xld-environment.yaml`

**Testing**
- [ ] Works on Linux, Windows and Mac
- [ ] Code is reviewed by someone else in the team
- [ ] Blueprint is manually tested with dockerised instances of our products
- [ ] CI is green

### CODE REVIEW

File: aws/lambda/blueprint.yaml [hunk 1]
```diff
@@ -0,0 +1,64 @@
+apiVersion: xl/v1
+kind: Blueprint
+metadata:
+  projectName: AWS Lambda and Credentials
+  description: |
+    This blueprint will configure one AWS Lamba application and one AWS credential set.
+  author: XebiaScrubs
```
Comment (deepu105) [inline aws/lambda/blueprint.yaml]: should be XebiaLabs as this is the official repo

File: aws/lambda/blueprint.yaml [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+  version: 1.0
+parameters:
+- name: AwsInfrastructureDirectory
+  type: Input
+  description: "Infrastructure directory (default: AmazonAWS)"
```
Comment (deepu105) [inline aws/lambda/blueprint.yaml]: Please stick to one template style, strings doesn't need `"` or `'` quote around in Yaml. I can see a mix of all here

File: aws/lambda/blueprint.yaml [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+  default: AmzAccount
+- name: AwsEnvironmentDirectory
+  type: Input
+  description: 'Environment directory (default: aws-staging)'
+  default: aws-staging
+- name: AwsEnvironmentName
+  type: Input
+  description: 'Environment name (default: "AWSHosts")'
+  default: AWSHosts
+- name: AwsAccessKeyID
```
Comment (deepu105) [inline aws/lambda/blueprint.yaml]: get these automatically, check the other AWS blueprints to see how its done https://github.com/xebialabs/blueprints/blob/development/aws/monolith/blueprint.yaml

File: aws/lambda/blueprint.yaml [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+  type: Input
+  description: AWS Access Key ID (example:AMZ1dF4Nz7)'
+- name: AwsAccessSecret
```
Comment (deepu105) [inline aws/lambda/blueprint.yaml]: same as above

File: aws/lambda/blueprint.yaml [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+  type: Input
+  description: 'AWS Region (example: wt9JK8VU5ue45)'
+- name: AwsRegion
```
Comment (deepu105) [inline aws/lambda/blueprint.yaml]: use a Select type and get this list using inbuilt function, see an existing aws blueprint for sample https://github.com/xebialabs/blueprints/blob/development/aws/monolith/blueprint.yaml

File: aws/lambda/blueprint.yaml [hunk 6]
```diff
@@ -0,0 +1,64 @@
...
+- name: AwsLambdaFunctionName
+  type: Input
+  description: 'Lambda function name (should match filename without extension) (default: function)'
+  default: function
+- name: AwsLambdaInitFunction
+  type: Input
+  description: 'Lambda main function (handler) (default: lambda_handler)'
+  default: lambda_handler
+- name: AwsLambdaRuntime
+  type: Input
```
Comment (deepu105) [inline aws/lambda/blueprint.yaml]: provide as Select type with options

File: aws/lambda/blueprint.yaml [hunk 7]
```diff
@@ -0,0 +1,64 @@
...
+  description: 'Lambda runtime (default: python3.7)'
+  default: python3.7
+- name: AwsRole
+  type: Input
+  description: "Lamba function\'s IAM role (example: arn:aws:iam::222333444:role/lambda_basic_execution)"
+
+files:
+- path: xld-aws-environment.yaml.tmpl
```
Comment (deepu105) [inline aws/lambda/blueprint.yaml]: put these files inside a `xebialabs` folder, that's the convention we follow

File: aws/lambda/xebialabs/.gitignore [hunk 8]
```diff
@@ -0,0 +1 @@
+secrets.xlvals
```
Comment (deepu105) [inline aws/lambda/xebialabs/.gitignore]: this file is not needed, plz remove

File: aws/lambda/xebialabs/values.xlvals [hunk 9]
```diff
@@ -0,0 +1 @@
+# This file includes all non-secret values, you can add variables here and then refer them with '!value' tag in YAML files
```
Comment (deepu105) [inline aws/lambda/xebialabs/values.xlvals]: this file is not needed, plz remove

**Comment by vpartington** [comment]
@xdanw Since you created this blueprint, we've updated the DOD (based on discussions with @amitmohleji and @jdewinne amongst others). Can you check your blueprint against the DOD and check the boxes where applicable?

**Comment by vpartington** [comment]
Hi Dan,

Are you still planning to update this PR? If not, can you close it?

Regards, Vincent.

**Pull request closed without merging**